### DISCUSSION TITLE
Supporting more than two parameters with arithmetic

### DISCUSSION THREAD

**Discussion by jaspertey** [tpetry/laravel-query-expressions#23] [Ideas]
I found myself needing to perform an `Add` on three columns and ended up having to do:

```php
new Add('treatments', new Add('assessments', 'no_shows'))
```

It would've been nice to do:

```php
new Add('treatments', 'assessments', 'no_shows')
```

In general, to be able to specify more than two parameters (variable-length argument) with the arithmetic expressions. Would this be a welcomed enhancement? Or would this compromise the ability to support different databases?

It would involve changing the `OperatorExpression`'s constructor (or introducing a `MultiOperatorExpression` since this wouldn't apply to all) to the effect of:

```php
public function __construct(string|Expression ...$values)
```

Leveraging PHP's "Type declared variable arguments".

Would love to hear your thoughts on this.

**Comment by tpetry** [comment]
Sounds like a good change 👍

But you can't change the `OperatorExpression` abstract class as the comparison operators wouldn't make sense with more than two arguments. Currently I did use arrays to indicate an endless amount of arguments. But the artihmetic and bitwise operators are special as they need at least two arguments. Do the bitwise ones work with many arguments?

I prefer a new interface with params like `__construct(string|Expression $value1, string|Expression $value2, string|Expression ...$values = [])` to make sure that the constructor is called with at least two values.

**Comment by jaspertey** [comment]
> Do the bitwise ones work with many arguments?

Yes, chaining multiple bitwise operators is mathematically valid... also checked on my local mariadb to confirm it doesn't complain.

Is this an update you'll be handling or is a PR welcome?

**Reply by tpetry** [reply]
a PR is welcome 😀

**Reply by tpetry** [reply]
@JasperTey Can you mark the discussion as answered as your PR has been merged?

**Reply by jaspertey** [reply]
Was a bit late seeing this, it got closed by you and no further action needed I take it?

**Reply by tpetry** [reply]
You pr-ed the change and its merged. So the discussion is solved by your feature request being implemented and available to use.
